cardano-cli/src/Cardano/CLI/Run.hs

```diff
@@ -53,6 +62,26 @@ renderClientCommandError (ByronClientError err) =
 renderClientCommandError (ShelleyClientError cmd err) =
   renderShelleyClientCmdError cmd err
 
+-- | Combine an 'ExceptT' that will write a warning message to @stderr@ with
+-- the provided 'ExceptT'.
+ioExceptTWithWarning :: MonadIO m => Text -> ExceptT e m () -> ExceptT e m ()
+ioExceptTWithWarning warningMsg e =
+  liftIO (Text.hPutStrLn stderr warningMsg) >> e
```

Comment:
I personally wouldn't bother with defining two new functions, seems verbose when 
`liftIO (Text.hPutStrLn stderr warningMsg) >> e` on its own would do.

Reply:
I was considering reusing this for other deprecated commands/arguments.

For example, after https://github.com/input-output-hk/cardano-node/pull/1979 is merged, I'm considering printing deprecation warnings when hex or `TextEnvelope` keys are specified.